File: python/src/apps/list_cameras.py
"""List available camera devices."""

from __future__ import annotations

import sys

from src.camera import ChessCamera


def run_list_cameras() -> int:
    devices = ChessCamera.list_devices()
    if not devices:
        print("No working cameras found.")
        print("Tips:")
        print("  - Plug in the USB webcam")
        print("  - Close other apps using the camera")
        print("  - Add your user to the video group: sudo usermod -aG video $USER")
        return 1

    print("Working cameras:")
    for device in devices:
        print(f"  {device['path']}: {device['resolution']}")
    return 0


def main() -> None:
    raise SystemExit(run_list_cameras())


if __name__ == "__main__":
    main()

File: apps/list_cameras.cpp
#include <iostream>

#include "chess_robot/camera/capture.hpp"
#include "common.hpp"

int main(int argc, char** argv) {
  using namespace chess_robot;
  using namespace chess_robot::apps;

  parseConfigArgs(argc, argv, "List available camera devices");
  const auto devices = ChessCamera::listDevices();
  if (devices.empty()) {
    std::cout << "No working cameras found.\n"
              << "Tips:\n"
              << "  - Plug in the USB webcam\n"
              << "  - Close other apps using the camera\n"
              << "  - Add your user to the video group: sudo usermod -aG video $USER\n";
    return 1;
  }

  std::cout << "Working cameras:\n";
  for (const DeviceInfo& device : devices) {
    std::cout << "  " << device.path << ": " << device.resolution << "\n";
  }
  return 0;
}
